### ISSUE TITLE
代码运行诸多报错

### ISSUE THREAD

**Issue by liangbaowen** [yyq19990828/DETR-series#2]
在成功编译算子之后，在运行train.py时就出现了个别方法名不正确，个别方法未定义，在修复这两个问题后又出现了'dict' object has no attribute 'model'问题，一个一个报错修复后紧接着会出现越来越多的错误，还有配置文件中的epochs错写成了epoches等问题，请问您是否使用此代码成功运行呢？还是说在上传代码时上传了错误的版本？

**Comment by Yshelgi** [comment]
> 在成功编译算子之后，在运行train.py时就出现了个别方法名不正确，个别方法未定义，在修复这两个问题后又出现了'dict' object has no attribute 'model'问题，一个一个报错修复后紧接着会出现越来越多的错误，还有配置文件中的epochs错写成了epoches等问题，请问您是否使用此代码成功运行呢？还是说在上传代码时上传了错误的版本？

好不容易看到一个pytorch实现，你这说得好害怕哈哈哈

**Comment by liangbaowen** [comment]
> > 在成功编译算子之后，在运行train.py时就出现了个别方法名不正确，个别方法未定义，在修复这两个问题后又出现了'dict' object has no attribute 'model'问题，一个一个报错修复后紧接着会出现越来越多的错误，还有配置文件中的epochs错写成了epoches等问题，请问您是否使用此代码成功运行呢？还是说在上传代码时上传了错误的版本？
> 
> 好不容易看到一个pytorch实现，你这说得好害怕哈哈哈

你下载来调试一下吧，我已经调试了两天了，实在找不到解决办法了，似乎是作者上传错了代码是个半成品，真的太多bug，没办法运行，要是你可以正确运行也教教我

**Comment by Yshelgi** [comment]
> > > 在成功编译算子之后，在运行train.py时就出现了个别方法名不正确，个别方法未定义，在修复这两个问题后又出现了'dict' object has no attribute 'model'问题，一个一个报错修复后紧接着会出现越来越多的错误，还有配置文件中的epochs错写成了epoches等问题，请问您是否使用此代码成功运行呢？还是说在上传代码时上传了错误的版本？
> > 
> > 
> > 好不容易看到一个pytorch实现，你这说得好害怕哈哈哈
> 
> 你下载来调试一下吧，我已经调试了两天了，实在找不到解决办法了，似乎是作者上传错了代码是个半成品，真的太多bug，没办法运行，要是你可以正确运行也教教我

试了一下，`workspace.py`里面create函数真的不知道怎么在处理配置参数，正打算重写读取配置那套

**Comment by liangbaowen** [comment]
> 试了一下，`workspace.py`里面create函数真的不知道怎么在处理配置参数，正打算重写读取配置那套

老哥加一下联系方式，13666913224@qq.com（发邮箱、加qq都可以），一块研究一下，我又改了很久，一点进展都没有，配置参数这块问题太多了

**Comment by liangbaowen** [comment]
> > > > 在成功编译算子之后，在运行train.py时就出现了个别方法名不正确，个别方法未定义，在修复这两个问题后又出现了'dict' object has no attribute 'model'问题，一个一个报错修复后紧接着会出现越来越多的错误，还有配置文件中的epochs错写成了epoches等问题，请问您是否使用此代码成功运行呢？还是说在上传代码时上传了错误的版本？
> > > 
> > > 
> > > 好不容易看到一个pytorch实现，你这说得好害怕哈哈哈
> > 
> > 
> > 你下载来调试一下吧，我已经调试了两天了，实在找不到解决办法了，似乎是作者上传错了代码是个半成品，真的太多bug，没办法运行，要是你可以正确运行也教教我
> 
> 试了一下，`workspace.py`里面create函数真的不知道怎么在处理配置参数，正打算重写读取配置那套

老哥成功跑起来了吗？

**Comment by Yshelgi** [comment]
> > > > > 在成功编译算子之后，在运行train.py时就出现了个别方法名不正确，个别方法未定义，在修复这两个问题后又出现了'dict' object has no attribute 'model'问题，一个一个报错修复后紧接着会出现越来越多的错误，还有配置文件中的epochs错写成了epoches等问题，请问您是否使用此代码成功运行呢？还是说在上传代码时上传了错误的版本？
> > > > 
> > > > 
> > > > 好不容易看到一个pytorch实现，你这说得好害怕哈哈哈
> > > 
> > > 
> > > 你下载来调试一下吧，我已经调试了两天了，实在找不到解决办法了，似乎是作者上传错了代码是个半成品，真的太多bug，没办法运行，要是你可以正确运行也教教我
> > 
> > 
> > 试了一下，`workspace.py`里面create函数真的不知道怎么在处理配置参数，正打算重写读取配置那套
> 
> 老哥成功跑起来了吗？

打算等出差回去有时间再弄

**Comment by yyq19990828** [comment]
sorry 啊xdm 我是在学习claude code用法的时候自己瞎尝试的, 后来工作太忙了 就一直没搞下去

**Comment by yyq19990828** [comment]
非常抱歉让你和参与讨论的朋友等了这么久，也抱歉早期版本让大家投入了很多无效调试时间。

你们当时的判断是对的：那一版确实是未完成的迁移，不是你们使用方式有问题。`workspace/config/trainer` 接口没有闭环、方法与配置项不一致等问题，会导致修完一个错误又暴露下一个错误；当时不应该把它当作可运行版本发布。

目前迁移已经重新推进，最新已提交的 `main` 不再以旧的根目录 `train.py` 和“逐个补洞”的方式作为主入口，而是使用 uv 环境和包内 CLI。已经实际验证：

- R18 config、完整模型、loss、backward、optimizer 和短训练链可运行；
- 2-GPU AMP 在真实 COCO train2017 完成一个完整 epoch，并完成 val2017，单 epoch 可运行性 checkpoint 的 `AP/AP50/AP75 = 0.468/0.643/0.504`；
- checkpoint 恢复、EMA、DDP sampler/RNG、梯度累积等链路已有测试和真实双卡 smoke；
- 最新已提交版本的默认回归为 `182 passed, 8 skipped`；
- 官方同权重 R18 Paddle/PyTorch 完整 val2017 的 AP 已对齐到 `1.65599e-7` 绝对差。

建议从最新 `main` 重新拉取，并从下面开始，而不是继续修旧 checkout：

```bash
uv sync --extra dev
.venv/bin/rtdetrv3-train --help
.venv/bin/rtdetrv3-eval --help
```

仓库仍不能声称“全部完成”：72 epoch 多 seed、R34/R50 完整训练、导出/部署和性能发布门槛仍未关闭。我们刚开了 #3，征集社区协助完成耗时很长的 R18 稳定性实验；统一运行脚本会在落到 `main` 后更新到该 Issue。

如果你在最新 `main` 仍能复现错误，请直接贴出：`git rev-parse HEAD`、Python/PyTorch/CUDA 版本、完整命令和第一个完整 traceback。不要再逐个手改报错，我会按当前版本的可复现路径处理。

再次向你和楼里一起排查的朋友道歉，确实让大家久等了。
